crates/vm2/src/heap.rs

```diff
@@ -69,55 +69,8 @@ impl Heap {
         Self { pages }
     }
 
-    /// Needed only by tracers
-    pub(crate) fn read_byte(&self, address: u32) -> u8 {
-        let (page, offset) = address_to_page_offset(address);
-        self.page(page).map(|page| page.0[offset]).unwrap_or(0)
-    }
-
-    fn page(&self, idx: usize) -> Option<&HeapPage> {
-        self.pages.get(idx)?.as_ref()
-    }
-
-    fn get_or_insert_page(&mut self, idx: usize, pagepool: &mut PagePool) -> &mut HeapPage {
-        if self.pages.len() <= idx {
-            self.pages.resize(idx + 1, None);
-        }
-        self.pages[idx].get_or_insert_with(|| pagepool.allocate_page())
-    }
-
-    fn write_u256(&mut self, start_address: u32, value: U256, pagepool: &mut PagePool) {
-        let (page_idx, offset_in_page) = address_to_page_offset(start_address);
-        let bytes_in_page = HEAP_PAGE_SIZE - offset_in_page;
-        let page = self.get_or_insert_page(page_idx, pagepool);
-
-        if bytes_in_page >= 32 {
-            value.to_big_endian(&mut page.0[offset_in_page..offset_in_page + 32]);
-        } else {
-            let mut bytes = [0; 32];
-            value.to_big_endian(&mut bytes);
-            let mut bytes_iter = bytes.into_iter();
-
-            for (dst, src) in page.0[offset_in_page..].iter_mut().zip(bytes_iter.by_ref()) {
-                *dst = src;
-            }
-
-            let page = self.get_or_insert_page(page_idx + 1, pagepool);
-            for (dst, src) in page.0.iter_mut().zip(bytes_iter) {
-                *dst = src;
-            }
-        }
-    }
-}
-
-#[inline(always)]
-fn address_to_page_offset(address: u32) -> (usize, usize) {
-    let offset = address as usize;
-    (offset >> 12, offset & (HEAP_PAGE_SIZE - 1))
-}
-
-impl HeapInterface for Heap {
-    fn read_u256(&self, start_address: u32) -> U256 {
+    // TODO: reduce visibility once `multivm` uses `StateInterface` APIs
```

Comment:
I plan to do this in a follow-up PR to keep the scope of this one reasonable.

Reply:
In theory, a lot of fields in state could be made private using this method as well.

Reply:
Yep, that's the plan.